lib/matplotlib/cm.py

```diff
@@ -255,10 +254,18 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
                     xx = x
                 else:
                     raise ValueError("third dimension must be 3 or 4")
-                if bytes and xx.dtype != np.uint8:
-                    xx = (xx * 255).astype(np.uint8)
-                if not bytes and xx.dtype == np.uint8:
-                    xx = xx.astype(float) / 255
+                if xx.dtype.kind == 'f':
+                    if xx.max() > 1 or xx.min() < 0:
```

Comment:
At least post-image-refactor these images are at "screen resolution" and not at data resolution which could be larger.